applications/StructuralMechanicsApplication/python_scripts/structural_mechanics_solver.py

```diff
@@ -66,6 +65,15 @@ def __init__(self, model, custom_settings):
             custom_settings["builder_and_solver_settings"].AddValue("use_block_builder", custom_settings["block_builder"])
             custom_settings.RemoveValue("block_builder")
 
+        settings_have_line_search = custom_settings.Has("line_search")
+        if settings_have_line_search:
+            kratos_utilities.IssueDeprecationWarning('MechanicalSolver', 'Using "line_search", please move it to "solving_strategy_settings" as "type"')
+            if not custom_settings.Has("solving_strategy_settings"):
+                custom_settings.AddEmptyValue("solving_strategy_settings")
+
+            custom_settings["solving_strategy_settings"].AddValue("type", "line_search")
```

Comment:
```suggestion
            custom_settings["solving_strategy_settings"].AddValue("type", custom_settings["line_search"])
```

otherwise will not work

Reply:
Not even that will work :-), as it is a string.